plugins/modules/s3_object.py

```diff
@@ -591,49 +604,73 @@
         s3.delete_object(Bucket=bucket, Key=obj)
         module.exit_json(msg="Object deleted from bucket %s." % (bucket), changed=True)
     except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
-        module.fail_json_aws(e, msg="Failed while trying to delete %s." % obj)
-
+        raise S3ObjectFailure("Failed while trying to delete %s." % obj, e)
 
-def create_dirkey(module, s3, bucket, obj, encrypt, expiry):
-    if module.check_mode:
-        module.exit_json(msg="PUT operation skipped - running in check mode", changed=True)
+def put_object_acl(module, s3, bucket, obj, params={}):
     try:
-        params = {'Bucket': bucket, 'Key': obj, 'Body': b''}
-        if encrypt:
-            params['ServerSideEncryption'] = module.params['encryption_mode']
-        if module.params['encryption_kms_key_id'] and module.params['encryption_mode'] == 'aws:kms':
-            params['SSEKMSKeyId'] = module.params['encryption_kms_key_id']
-
-        s3.put_object(**params)
+        if params:
+            s3.put_object(**params)
         for acl in module.params.get('permission'):
             s3.put_object_acl(ACL=acl, Bucket=bucket, Key=obj)
-    except is_boto3_error_code(IGNORE_S3_DROP_IN_EXCEPTIONS):
-        module.warn("PutObjectAcl is not implemented by your storage provider. Set the permissions parameters to the empty list to avoid this warning")
     except is_boto3_error_code('AccessControlListNotSupported'):
         module.warn("PutObjectAcl operation : The bucket does not allow ACLs.")
     except (botocore.exceptions.BotoCoreError, botocore.exceptions.ClientError) as e:  # pylint: disable=duplicate-except
-        module.fail_json_aws(e, msg="Failed while creating object %s." % obj)
+        raise S3ObjectFailure("Failed while creating object %s." % obj, e)
+
 
+def create_dirkey(module, s3, bucket, obj, encrypt, expiry):
+    if module.check_mode:
+        module.exit_json(msg="PUT operation skipped - running in check mode", changed=True)
+    
+    params = {'Bucket': bucket, 'Key': obj, 'Body': b''}
+    params.update(get_extra_params(module, encrypt))
+    put_object_acl(module, s3, bucket, obj, params)
+        
     # Tags
     tags, _changed = ensure_tags(s3, module, bucket, obj)
```

Comment:
;;; Type C-c C-c to include your comment locally
;;; After that check the transient menu (press r)i to see possible actions
Trailing whitespaces

Reply:
r;; Type C-c C-c to include your comment locally
;;; After that check the transient menu (press r) to see possible actions

Reply:
a